### PR TITLE
*: fix AnalyzePartitionMergeConcurrency unavailable

### LINKED ISSUE #47594: AnalyzePartitionMergeConcurrency cannot be abled
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

set tidb_merge_partition_stats_concurrency=10;

the code of tidb_merge_partition_stats_concurrency never is covered.

### 2. What did you expect to see? (Required)

we will create 10 worker to do it.
### 3. What did you see instead (Required)

we will create 1 worker to do it.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #47594

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handletest/globalstats/globalstats_test.go [hunk 1]
```diff
@@ -1127,6 +1128,8 @@ func TestGlobalIndexStatistics(t *testing.T) {
 	h.SetLease(time.Millisecond)
 	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
+	sctx := tk.Session().(sessionctx.Context)
+	sctx.GetSessionVars().AnalyzePartitionMergeConcurrency = analyzePartitionMergeConcurrency
```
Comment (elsa0520) [inline statistics/handle/handletest/globalstats/globalstats_test.go:1132]: This test doesn't seem to verify that the concurrency settings are set successfully?

Comment (hawkingrei) [reply statistics/handle/handletest/globalstats/globalstats_test.go]: Yes, It is my problem.

**Review by qw4990 — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**